service/frontend/service.go

```diff
@@ -200,7 +200,7 @@ func (s *Service) Start() {
 	}
 
 	metricsBlobstore := blobstore.NewMetricClient(params.BlobstoreClient, base.GetMetricsClient())
-	wfHandler := NewWorkflowHandler(base, s.config, metadata, history, historyV2, visibility, kafkaProducer, metricsBlobstore)
+	wfHandler := NewWorkflowHandler(base, s.config, metadata, history, historyV2, visibility, kafkaProducer, metricsBlobstore, params.HistoryArchivers, params.VisibilityArchivers)
```

Comment:
Another idea is adding these two variables to serviceImpl and add new methods to the service interface to return them.

Reply:
I like the current approach. It seems like they should follow the same approach as existing items like kafkaProducer.